meta-lmp-base/recipes-kernel/xenomai4/xenomai4_git.bb

```diff
@@ -0,0 +1,50 @@
+SUMMARY = "Xenomai 4"
+DESCRIPTION = "Xenomai 4 user support (libevl)"
+LICENSE = "MIT"
+LIC_FILES_CHKSUM = "file://LICENSE;md5=2500c28da9dacbee31a3f4d4f69b74d0"
+
+SRCBRANCH = "master"
+
+SRC_URI = "git://source.denx.de/Xenomai/xenomai4/libevl.git;protocol=https;branch=${SRCBRANCH}"
+SRCREV = "dbb1a86f011d7b87f1420fbf13753ea9fccef6af"
+
+S = "${WORKDIR}/git"
+
+TOOLCHAIN = "gcc"
+
+# We need the kernel sources to be available, so build it first.
+DEPENDS += "virtual/kernel"
+
+# We may need this for 64bit atomic ops.
+RDEPENDS:${PN} = "libgcc"
+
+inherit pkgconfig meson
+
+PACKAGECONFIG[enable-debug] = "-Doptimization=0 -Ddebug=true,,,"
+
+do_compile[depends] += "virtual/kernel:do_deploy"
+EXTRA_OEMESON = "-Duapi=${STAGING_KERNEL_DIR}"
+
+FILES:${PN} = "\
+    ${bindir}/evl \
+    ${bindir}/latmus \
+    ${bindir}/hectic \
+    ${bindir}/oob-spi \
+    ${bindir}/oob-net-icmp \
+    ${libdir}/libevl.so.* \
+    ${libexecdir}/evl \
+    ${prefix}/tests \
+    "
+
+FILES:${PN}-dev = "\
+    ${libdir}/libevl.so \
+    ${libdir}/pkgconfig/evl.pc \
+    ${includedir}/evl \
+    ${includedir}/uapi/evl \
+    "
+
+FILES:${PN}-dbg = "\
+    ${libdir}/.debug/libevl.so \
+    "
```

Comment:
This block [1] where we specify all the files can be replaced with a new package that install the tests and the rest goes in the default packages:
```
-FILES:${PN} = "\
-    ${bindir}/evl \
-    ${bindir}/latmus \
-    ${bindir}/hectic \
-    ${bindir}/oob-spi \
-    ${bindir}/oob-net-icmp \
-    ${libdir}/libevl.so.* \
-    ${libexecdir}/evl \
-    ${prefix}/tests \
-    "
-
-FILES:${PN}-dev = "\
-    ${libdir}/libevl.so \
-    ${libdir}/pkgconfig/evl.pc \
-    ${includedir}/evl \
-    ${includedir}/uapi/evl \
-    "
-
-FILES:${PN}-dbg = "\
-    ${libdir}/.debug/libevl.so \
-    "
+PACKAGES += "${PN}-test"
+
+FILES:${PN}-test = "${prefix}/tests"
+
+# install the test by default and use BAD_RECOMMENDATIONS when not desired
+RRECOMMENDS:${PN} += "${PN}-test"
```

Reply:
I dont understand this bit but will just push it - I assume you have tested it?

Reply:
Yeah, I do some tests locally.
My suggestion is because the files installed to `${prefix}/tests` is to test and check at runtime if the xenomai was installed and configured correctly so in productions it won't be necessary.

This will create two base packages `xenomai4.ipk` and `xenomai4-test.ipk` with the second being a dependency of the first. When in production `BAD_RECOMMENDATIONS =+ "xenomai4-test"` will only install the first removing the dependence on the test package.